database-commons/src/main/java/co/cask/ConnectionConfig.java

```diff
@@ -32,20 +33,27 @@
 /**
  * Defines a base {@link PluginConfig} that Database source, sink, and action can all re-use.
  */
-public class ConnectionConfig extends PluginConfig {
+public abstract class ConnectionConfig extends PluginConfig {
   public static final String CONNECTION_STRING = "connectionString";
   public static final String USER = "user";
+  public static final String HOST = "host";
+  public static final String PORT = "port";
+  public static final String DATABASE = "database";
   public static final String PASSWORD = "password";
   public static final String CONNECTION_ARGUMENTS = "connectionArguments";
-  public static final String JDBC_PLUGIN_NAME = "jdbcPluginName";
-  public static final String JDBC_PLUGIN_TYPE = "jdbcPluginType";
-  public static final String COLUMN_NAME_CASE = "columnNameCase";
-  public static final String ENABLE_AUTO_COMMIT = "enableAutoCommit";
+  public static final String JDBC_PLUGIN_TYPE = "jdbc";
 
-  @Name(CONNECTION_STRING)
-  @Description("JDBC connection string including database name.")
-  @Macro
-  public String connectionString;
```

Comment:
The generic DB plugins should still require the user to configure the connection string, since JDBC drivers can do whatever they want with the URL. It's only the DB specific plugins that should expose the separate properties for host, port, db, etc.